src/main/java/io/github/balconyseats/vertx/helper/kafka/KafkaHelper.java

```diff
@@ -0,0 +1,170 @@
+package io.github.balconyseats.vertx.helper.kafka;
+
+import io.github.balconyseats.vertx.helper.exception.IllegalConfigurationException;
+import io.github.balconyseats.vertx.helper.util.ConfigUtil;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaConsumer;
+import io.vertx.kafka.client.producer.KafkaProducer;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Helper to create consumer and producer from configuration
+ * <pre>
+ * kafka:
+ *   enabled: true
+ *   consumers:
+ *     commonConfig:
+ *       key.deserializer : 'org.apache.kafka.common.serialization.StringDeserializer'
+ *       value.deserializer : 'org.apache.kafka.common.serialization.StringDeserializer'
+ *     foo:
+ *       client.id: foo
+ *       topic: foo_topic
+ *       group.id: foo
+ *     bar:
+ *       client.id: bar
+ *       topic: bar_topic
+ *       group.id: bar
+ *   producers:
+ *     commonConfig:
+ *       key.serializer : 'org.apache.kafka.common.serialization.StringSerializer'
+ *       value.serializer : 'org.apache.kafka.common.serialization.StringSerializer'
+ *     foo:
+ *       client.id: foo
+ *       topic: foo
+ *     bar:
+ *       client.id: bar
+ *       topic: bar
+ *   config:
+ *     bootstrap.servers: 'server-1:9092,server-2:9092'
+ *     security.protocol : 'SSL'
+ *     ssl.enabled.protocols: 'TLSv1.2,TLSv1.1,TLSv1'
+ *     ssl.keystore.location: 'keystore.jks'
+ *     ssl.keystore.password: 'changeit'
+ *     ssl.keystore.type: 'JKS'
+ *     ssl.truststore.location: 'truststore.jks'
+ *     ssl.truststore.password: 'password'
+ *     ssl.truststore.type: 'JKS'
+ * </pre>
+ */
+public class KafkaHelper {
+
+    /**
+     * Is kafka enabled in configuration 'kafka.enabled' property
+     * @param config configuration
+     * @return true if is enabled or false if not
+     */
+    public static boolean isKafkaEnabled(JsonObject config) {
+        return ConfigUtil.getBoolean("/kafka/enabled", config);
+    }
+
+    /**
+     * Creates {@link KafkaConsumer} from config object
+     * @param vertx vertx instance
+     * @param consumerName consumer name in config
+     * @param config configuration
+     * @param <K> key type
+     * @param <V> value type
+     * @return {@link KafkaConsumer} instance
+     */
+    public static <K, V> KafkaConsumer<K, V> createConsumer(Vertx vertx, String consumerName, JsonObject config) {
+
+        JsonObject kafkaConfig = kafkaConfig(config);
+
+        Map<String, String> commonKafkaConfig = commonKafkaConfig(kafkaConfig);
+        Map<String, String> consumerConfig = consumerConfig(consumerName, kafkaConfig);
+
+        consumerConfig.putAll(commonKafkaConfig);
```

Comment:
Is it possible for a specific consumer to override common config?
Then specific consumer config should be put into map **after** common config.

Reply:
Maybe to add method whic accept custom config and merge it in this config.
`createConsumer(Vertx vertx, String consumerName, JsonObject config, Map<String, String> customConfig)`

Reply:
The question/note is about the following possibility:

```
kafka:
  enabled: true
  consumers:
    commonConfig:
      key.deserializer: 'org.apache.kafka.common.serialization.StringDeserializer'
      value.deserializer: 'org.apache.kafka.common.serialization.StringDeserializer'
    communication-kernel:
      client.id: communication-engine-gateway.consumer
      topic: communication-kernel-output
      group.id: communication-engine-gateway
      bootstrap.servers: some.host:9093
  config:
    bootstrap.servers: localhost:9092
```

and `consumerConfig.putAll(commonKafkaConfig);` will overwrite specific consumer config with common kafka config.
It should be vice versa.